src/components/Header/HamburgerBotton.tsx

```diff
@@ -0,0 +1,109 @@
+import {
+  SITEMAP,
+  registerAndLoginList,
+  userList,
+} from '@components/Header/menu';
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from '@components/ui/popover';
+import { Separator } from '@components/ui/separator';
+import { Menu, X } from 'lucide-react';
+import Image from 'next/image';
+import Link from 'next/link';
+import { useState } from 'react';
+
+type Props = {
+  isLoggedin: boolean;
+  onSignOut: () => void;
+};
+
+type List = {
+  name: string;
+  icon?: JSX.Element;
+  url: string;
+};
+
+const defaultAvatar = '/header__defaultAvatar.svg';
+const fakeAvatar = '/header__fakeAvatar.svg';
+
+const HamburgerBotton = ({ isLoggedin, onSignOut }: Props) => {
+  const [isOpen, setIsOpen] = useState(false);
+
+  return (
+    <Popover open={isOpen} onOpenChange={setIsOpen}>
+      <PopoverTrigger className="mx-[3px] flex items-center justify-center rounded-full border border-primary p-3 data-[state=open]:mx-0 data-[state=open]:border-4">
+        {isOpen ? <X size={24} /> : <Menu size={24} />}
+        <Image
+          className="ml-2"
+          src={isLoggedin ? fakeAvatar : defaultAvatar}
+          alt="user"
+          width={40}
+          height={40}
+        />
+      </PopoverTrigger>
+      <PopoverContent className="h-0 w-0 border-none bg-transparent p-0">
+        <div className="no-scrollbar absolute right-[-56px] top-4 hidden h-fit max-h-[450px] w-[272px] overflow-scroll rounded-[32px] border-4 border-black bg-surface pt-6 xl:block">
+          {isLoggedin ? (
+            <>
+              {userList.map((user: List) => (
+                <Link
+                  className="mb-4 flex justify-between px-8 py-2"
+                  key={user.name}
+                  href={user.url}
+                >
+                  <div className="text-xl font-semibold">{user.name}</div>
+                  {user.icon && user.icon}
+                </Link>
+              ))}
+            </>
+          ) : (
+            <>
+              {registerAndLoginList.map((list: List) => (
```

Comment:
hover effect is missing

https://www.figma.com/design/6JPbEOj0WIGCAr3gJQU0Cb/%E5%85%AD%E8%A7%92_ChillKa?node-id=277-3319&t=sWspYS20Tmn6J19y-4